tests/integration/helm/install_test.go

```diff
@@ -0,0 +1,220 @@
+// +build integ
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	kubeapicore "k8s.io/api/core/v1"
+	kubeapimeta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/image"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/helm"
+	kubetest "istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/util/sanitycheck"
+	"istio.io/pkg/log"
+)
+
+const (
+	IstioNamespace      = "istio-system"
+	ReleasePrefix       = "istio-"
+	BaseChart           = "base"
+	DiscoveryChart      = "istio-discovery"
+	IngressGatewayChart = "istio-ingress"
+	EgressGatewayChart  = "istio-egress"
+	BaseReleaseName     = ReleasePrefix + BaseChart
+	IstiodReleaseName   = "istiod"
+	IngressReleaseName  = IngressGatewayChart
+	EgressReleaseName   = EgressGatewayChart
+	ControlChartsDir    = "istio-control"
+	GatewayChartsDir    = "gateways"
+	retryDelay          = 2 * time.Second
+	retryTimeOut        = 20 * time.Minute
```

Comment:
this timer is too long. The CI has a 10 minute timer on no output. Typically we would run this timer at 300 seconds.

Reply:
Done